arroyo-controller/src/schedulers/mod.rs

```diff
@@ -378,25 +386,28 @@ impl NodeScheduler {
             worker_id = worker_id.0
         );
 
-        let mut client = NodeGrpcClient::connect(format!("http://{}", node.addr)).await?;
-
-        match (
-            client
-                .stop_worker(Request::new(StopWorkerReq {
-                    job_id: job_id.to_string(),
-                    worker_id: worker_id.0,
-                    force,
-                }))
-                .await?
-                .get_ref()
-                .status(),
-            force,
-        ) {
+        let Ok(mut client) = NodeGrpcClient::connect(format!("http://{}", node.addr)).await else {
+            warn!("Failed to connect to worker to stop; this likely means it is dead");
```

Comment:
Yeah, I think there are ways of making this more robust but waiting for a heartbeat does mean taking longer to recover failed pipelines.

There are basically two cases:
* The node process has died, in which case the workers will also have died (generally) because they are child processes of the node
* We have a network partition between the controller and the node

In case 1, we're doing the right thing. In case 2, the worker and the node will each die on their own as they fail to talk to the controller.